[IV 21-22] Objetivo 2

NachoCarher opened the pull request:
Pull request para el objetivo 2 de la asignatura IV.
Se ha avanzado con la HU1 #3 correspondiente al M0. Y se solucionan los issues relacionados con la expansión de las características de las series y su valoración. #15 #8 #7

NachoCarher committed:
Implementacion de la estructura de datos para M0

Se ha creado la estructura de datos para una serie y se avanza en la solución de [Tiene que haber una manera de identificar las series de forma única](#7) asociada a HU1(#3)

NachoCarher committed:
Segunda versión de la implementación de clases

La valoración no es considerada como una característica fija preestablecida de una serie, si no un atributo de la propia clase serie que va se va actualizando la media a medida que los usuarios introducen nuevas entradas. Este cambio está ligado con el [issue](#15) asociado a la [HU1](#3)

NachoCarher committed:
Tercera versión de la implementación de clases

Las series deben contar con unas características por las que podrán ser filtradas. Estos cambios cubren #8 que está asociado a las HUs #3 y #5

JJ commented:
No has hecho un PR a la asignatura.

JJ commented:
Y por favor, haz correctamente los mensajes de commit.

NachoCarher commented:
Ya he hecho PR a la asignatura

NachoCarher commented:
@Parka015 para que el PR estuviese completo al 100% sería necesario que le asignases los issues en los que se trabaja y el milestone al que se asocian, ya que es algo que no puedo modificar yo. Está todo referenciado en los commits realizados.

Parka015 commented:
> @Parka015 para que el PR estuviese completo al 100% sería necesario que le asignases los issues en los que se trabaja y el milestone al que se asocian, ya que es algo que no puedo modificar yo. Está todo referenciado en los commits realizados.

Por alguna razón, no me deja, esta mencionado dentro del issue #7 que ese es el issue sobre el que tiene que trabajar, prueba a hacer un comentario en dicho issue para ver si no me deja asignar por eso

Parka015 reviewed (changes requested):
El issue #7  es el único del que debes encargarte, por lo tanto las demás issues y el archivo de características.py ya lo estoy desarrollando yo.
Además, debería aparecer un objeto de la clase características en la clase serie, ya que dichas características van asociadas a una serie.

Parka015 commented:
@NachoCarher aunque ya lo he corregido, recuerda asignar los PR  que hagas a un PMV . Si no sabes como se hace es al principio de la pagina del PR (esta por ejemplo) en el lado derecho donde viene Assignees, labels, Projects, **Milestone** ... Le pinchas y selecciones el Milestone.
Este PR en principio esta asociado al M0 asi que lo dejo así

NachoCarher commented:
@Parka015 sé que hay que asignar los PR a un issue y milestone, pero en este caso, como el PR está siendo a tu repo, no es tarea mía realizar esa asignación. Básicamente porque no me deja, sólo tú tienes autorización para hacerlo, y por eso te he pedido que los asignes. Ya he visto que lo has actualizado. Gracias.

JJ commented on source/caracteristicas.py:
```diff
@@ -5,55 +5,72 @@
+    def getGenero(self):
+        gen = ""
+        
+        if self.genero == 1.0
```
Esta no es la mejor forma de hacer esto. Y en todo caso, no hay que meter código en este objetivo.

NachoCarher replied:
Se quería que los géneros fuesen valores reales y aquí implementé una correlación entre el nombre del género y su valor. En su defecto, no implemento código y lo dejo como estaba, tan solo con el return de su valor original.

JJ reviewed (commented):
Independientemente de lo que diga @Parka015 , deberías darle una vuelta a cómo implementar los issues que se han creado.

Parka015 commented:
> @Parka015 sé que hay que asignar los PR a un issue y milestone, pero en este caso, como el PR está siendo a tu repo, no es tarea mía realizar esa asignación. Básicamente porque no me deja, sólo tú tienes autorización para hacerlo, y por eso te he pedido que los asignes. Ya he visto que lo has actualizado. Gracias.

No sabía que no te dejaba, fallo mío entonces

NachoCarher committed:
Actualizo carcacteristicas.py

Se incluye la valoración en esta clase y se dejan las caracteristicas iniciales que tan sólo eran los géneros de las series. Avanza #3

NachoCarher committed:
Actualizo serie.py

La valoración está incluida en caracteristicas.py y se implementa el set para el identificador único de una serie. Avanza #3. Closes #7

NachoCarher committed:
Merge branch 'main' into Objetivo-2

JJ commented:
Los mensajes de commit nunca son "actualizo no sé qué" Di lo que haces y por qué.

JJ commented on source/serie.py:
```diff
@@ -1,2 +1,18 @@
+# Clase que almacena la información correspondiente a una serie
 
-#Por desarrollar (Nacho)
+Class serie:
```
No entiendo muy bien. Al final, ¿esto es todo lo que has hecho?

JJ reviewed (changes requested):
Después del último merge no tengo muy claro qué has hecho y como una clase que sólo tiene un ID trata de identificar el éxito de una serie.

NachoCarher commented:
@JJ Se soluciona el issue en el que se indica que una serie debe de tener un identificador único. Y el éxito de una serie se podrá predecir en base a los atributos que hay ya en la clase características. Toda la estructura de clases inicial ya está creada y el fichero .yaml tiene especificadas las entidades objeto/valor. No considero que sea necesario implementar más cosas y se avanza en la consecución de #3 y de #7.

JJ commented:
> @JJ Se soluciona el issue en el que se indica que una serie debe de tener un identificador único. Y el éxito de una serie se podrá predecir en base a los atributos que hay ya en la clase características. Toda la estructura de clases inicial ya está creada y el fichero .yaml tiene especificadas las entidades objeto/valor. No considero que sea necesario implementar más cosas y se avanza en la consecución de #3 y de #7.

¿Creando una clase con un ID? De veras, tendrás que currártelo un poquitín más. O haces alguna clase más, o clarificas qué necesita más esa clase, o haces algo que puedas usar en la lógica de negocio. Tienes varias opciones más para avanzar en el PMV.

NachoCarher committed:
Implementada la predicción del éxito de una serie.

Para alcanzar #3 es necesario poder identificar el éxito de una serie. Para ello se importa el módulo en el que se desarrolla el algoritmo predictivo en base a las características de ésta. Una vez realizadas las operaciones necesarias y calculada la estimación de la valoración, se guardará como un atributo válido (inicialmente nulo) de una serie.

NachoCarher commented:
@JJ he tratado de esclarecer aún más que era necesario para la clase serie.py. Para alcanzar #3 se requiere calcular por medio de un algoritmo determinado y desarrollado en otro módulo la predicción de la valoración de una serie. Por lo que esta estimación obtenida en función de las características es un atributo de cada serie y se puede asignar rápidamente mediante el identificador único previamente implementado.

JJ commented on source/caracteristicas.py:
```diff
@@ -22,16 +22,17 @@ def __init__(self,accion : float,terror : float,romance : float,\
         
     ####  Gets  ####
         
+
```
De veras, no hagáis esto. Genera ruido en la corrección y no sirve para nada.

JJ commented on source/serie.py:
```diff
@@ -1,2 +1,26 @@
+Class serie:
+  def __init__(self, iden):
+    self.identificador = iden   # Codigo para identificar de forma unica cada serie
+    self.prediccion_valoracion = -1 # Éxito estimado de una serie. Inicialmente nulo
```
De veras que una serie sólo tiene un identificador y una predicción de la valoración? ¿Nada más? ¿No tiene, no sé, el nombre de la serie o algo de eso? ¿Por qué importas KNN si no lo usas?

JJ reviewed (changes requested):
De veras, tendrás que hacer algo más si quieres superar el objetivo. NO se trata de que haya un mínimo de código a desarrollar, se trata de que es imposible que una serie sólo necesite eso.
Adicionalmente, usa las mejores prácticas de Python en los identificadores.

NachoCarher committed:
Se implementan más atributos de las series

Con el módulo KNN se calcula una predicción de la valoración de una serie sobre sus características y una vez calculada se utiliza el respectivo método de esta clase para asignarla. No era necesario importar el módulo.

Las series necesitan tener más características a parte del género (en caracteristicas.py) para poder distinguirlas y filtrarlas correctamente. Por ello se han implementado las básicas que tendría una serie. #3

NachoCarher committed:
Actualizo el identificador de la clase serie

NachoCarher committed:
Implementada clase catalogo para organizar series

Se implementa una clase que almacena una coleccion de objetos de la clase Serie. Facilita la organizacion de las series y el acceso a éstas.
Se agregan los métodos agregar, mostrar y eliminar correspondientes a la lista de series. No se ha escrito código dentro de estos métodos para una posterior implementación, sin embargo, con el uso de 'pass' permitimos que el código sea válido y se pueda ejecutar. #3

NachoCarher commented:
@JJ he solucionado lo respectivo al uso del import del módulo que no se utiliza. Las series es cierto que deben de contar con más atributos para poder trabajar con ellas correctamente y ser más fieles a la realidad, de manera que he implementado los esenciales. Y para facilitar el trabajo con varios objetos de la clase serie considero que es de gran utilidad contar con un catálogo que almacene todas ellas.

JJ commented on source/catalogo.py:
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Clase que almacena una coleccion de series
+
+Class Catalogo:
```
De veras, ni siquiera estás intentando respetar la sintaxis de Python. Por no mencionar que tampoco añades código usando issues.

JJ reviewed (changes requested):
Lo siento, no pasa este objetivo.

NachoCarher committed:
Corregidos errores de sintaxis y eliminado código ambig

NachoCarher committed:
Se soluciona un error de sintaxis

Se corrige un error de sintaxis que impedía la ejecución del programa y por tanto los catálogos de series no se podían generar #23 #3

NachoCarher commented:
@JJ el código añadido en los commits anteriores en relación al catálogo usa el issue #23, el resto de modificaciones han trabajado sobre #7 y #8, y todo ello para avanzar en la consecución de #3 del M0. Y además ya no hay errores de sintaxis. 

Creo que ya está todo lo suficiente para completar este objetivo.

JJ commented:
Lo siento, @NachoCarher ,pero creo que dejé bien claro que una vez que se corrigieran los objetivos después del periodo lectivo, no se podían reenviar. El objetivo está no superado, como puedes ver en el fichero correspondiente. Puedes proceder a los siguientes.

NachoCarher commented:
@Parka015 Podrías darme tu feedback sobre la última modificación que hice por favor. Todo los implementado respeta la sintaxis de Python y se trabaja siempre sobre un issue aunque antes no los referenciaba, y ya es algo que siempre estoy teniendo en cuenta para los otros objetivos. Gracias.

NachoCarher commented:
@JJ como le comenté por Telegram, salvo los commits, creo que todo está corregido y se puede revisar

Parka015 commented:
> @Parka015 Podrías darme tu feedback sobre la última modificación que hice por favor. Todo los implementado respeta la sintaxis de Python y se trabaja siempre sobre un issue aunque antes no los referenciaba, y ya es algo que siempre estoy teniendo en cuenta para los otros objetivos. Gracias.

Veo que has mejorado en el aspecto de los commits, que explicas que hacen y los asocias a un issue.
En cuanto a lo realizado creo que hay atributos más que suficientes para la clase Serie pero sigue habiendo el mismo problema que te comenté hace tiempo, "debería aparecer un objeto de la clase características en la clase serie".
En lugar de un atributo predicción usa un atributo características

NachoCarher committed:
Actualiza serie.py con atributo caracteristicas closes #8

Tal y como comentó @Parka015, una serie debería de contar con un atributo
que consistiese en un objeto de la clase caracteristicas.

Se añade un objeto de este tipo en lugar del atributo 'predicción'.

NachoCarher commented:
@Parka015 ya he solucionado el problema de los atributos que me comentaste. Gracias

JJ commented on source/catalogo.py:
```diff
@@ -0,0 +1,16 @@
+
+class Catalogo:
+  
+  def __init__(self, series = []):
```
Ninguna parte de este código está asociada a ningún issue. Algunas partes están relacionadas con una HU que habla de "predicción". No tiene nada que ver con agregar o mostrar o eliminar. ¿Cómo identificas una serie? ¿Qué características tiene cada una?

NachoCarher replied:
Le pregunté que si debido a no hacer los commits correctamente era motivo suficiente como para repetir el Objetivo. Y concluimos que no era necesario pues en los objetivos futuros demuestro que ya los hago siguiendo las buenas prácticas y son correctos.

Las caracteristicas de una serie están en el atributo características que corresponde a la clase Características y determinamos entre los dos que debia figurar así (desarrollado por @Parka015).

Cada serie cuenta con un identificador único definido en su clase correspondiente. Lo comentado no concierne a la clase Catálogo (asociada a #23), si no a la clase Serie.

JJ commented on source/serie.py:
```diff
@@ -1,2 +1,43 @@
+# Clase que almacena la información correspondiente a una serie
 
-#Por desarrollar (Nacho)
+from source.caracteristicas import Caracteristicas
```
Este no es un nombre que sigua las mejores prácticas. "source" no es un nombre correcto para un módulo, y hay que configurar el repo de forma que el nombre del módulo corresponda al del proyecto.

NachoCarher replied:
source es el nombre del directorio en el que se guardan los fuentes, y yo no tomé la decisión de llamarlo así. De hecho en mi repo no figura con ese nombre. 

Me he encargado de referenciar a ese módulo porque es lo que me solicitó Pablo en su último comentario.

JJ commented on source/serie.py:
```diff
@@ -1,2 +1,43 @@
+
+class Serie:
+  def __init__(self, iden, titulo, director, duracion, lanzamiento):
+    self.identificador = iden                             # Codigo para identificar de forma unica cada serie
```
es mala práctica usar nombres diferentes para el atributo y el argumento.
Además, hay que usar la práctica común para indicar si son privados o no.

NachoCarher replied:
> Lo siento, no superas este objetivo. Habría sido conveniente que lo compartieras con la clase para que te ayudaran, o simplemente que me hicieras caso a los comentarios anteriores.

Profesor, precisamente he procurado ir cumpliendo con los objetivos marcados en base a sus comentarios y recomendaciones. No sólo eso, sino que además en clase le he ido preguntando y me he interesado por entender sus pautas en reiteradas ocasiones. Y he tratado de plasmarlas lo mejor posible teniendo en cuenta el esfuerzo que ha depositado por que las entendiera y adaptándome en todo momento a ellas.

Llevo bloqueado aquí por meses tratando de documentarme al máximo para poder pasarlo y contrastándolo con el de los compañeros. Además de haber estado hablando con Pablo sobre cómo proceder y qué implementar.

A mi parecer, y sin tratar de quitarle importancia al objetivo, el cómo he desarrollado los siguientes a este demuestra qué sé lo suficiente como para pasarlo y rogaría lo tuviese más en consideración pues sé que es atento y tiene esos aspectos en cuenta, pues imagino que le dará gran importancia a que el alumno demuestre una buena evolución.

JJ commented on source/serie.py:
```diff
@@ -1,2 +1,43 @@
+  def getDuracion(self):
+    return self.duracion
+  
+  def getLanzamiento(self):
```
Repetidamente hemos dicho en clase que *no* se deben poner getters y setters si no lo pide la lógica de negocio. Tampoco está justificado. Salvo alguna línea, el resto simplemente no corresponden a ningún issue. Tampoco has hecho nada nuevo para la convocatoria extraordinaria.

NachoCarher replied:
@JJ de cara a la convocatoria extraordinaria trabajé en otra clase diferente bajo un issue asociado y habiendo leído toda la documentación sobre el hito, además de haber solucionado los errores destacados en la ordinaria e implementando lo que me solicitó Pablo en su comentario. 

Entiendo perfectamente lo de los getters y setters y el nombre y ámbito de los atributos, pero son puntualizaciones de los buenos procedimientos con el lenguaje de programación y los tendré en consideración. Profesor, he desarrollado lo suficiente e invertido en el objetivo las horas necesarios como para entenderlo y poder continuar con los siguientes.

JJ reviewed (changes requested):
Lo siento, no superas este objetivo. Habría sido conveniente que lo compartieras con la clase para que te ayudaran, o simplemente que me hicieras caso a los comentarios anteriores.
